### PR TITLE
Release 4.0.0

### PR DESCRIPTION
**Pull request by mikemorency**
##### SUMMARY
This is release 4.0.0, which includes new openapi specs for vsphere 8.0.2. It also includes two new modules:
- content_library_info - returns information about an existing content library
- content_library_subscriptions_info - returns information about subscriptions to an existing content library

In addition to the updated modules, the following is included in the release:
1. remove support for ansible-core 2.14
2. Up to date instructions and playbooks for generating module content in this repo. This information has been added to `development.md`
3. various doc updates for lookup plugins and the README

##### ADDITIONAL INFORMATION
Related: https://github.com/ansible-community/ansible.content_builder/pull/89

### CODE REVIEW

**Review by mariolenz — CHANGES_REQUESTED** [review]
I suggest to close #495 and drop ansible 2.14 in _this_ PR. That is:

1. Drop 2.14 from `.github/workflows/sanity-tests.yaml`
2. Update `meta/runtime.yaml` to require ansible >= 2.15.0
3. Delete `tests/sanity/ignore-2.14.txt`
4. Regenerate everything
5. Update README if necessary (I'm not sure if ansible.content_builder will do it automatically)

**Comment by mariolenz** [comment]
@mikemorency Maybe we need to define the `vmware.vmware` collection as a test dependency to make the docs test work?

**Comment by mikemorency** [comment]
do we want to do that though? its not an actual dependency

**Comment by mariolenz** [comment]
> do we want to do that though? its not an actual dependency

No, we don't want to make this a dependency for the collection if it isn't. I'm looking for a way to make it a dependency just for the docs test. Maybe using [extra-collections](https://github.com/ansible-community/github-docs-build/blob/8fd31e0a678d02f65ff56b5f7e75dad50f8a9f70/.github/workflows/_shared-docs-build-pr.yml#L161-L164) would help? :thinking: 

I don't know if this works if we do it in this PR, though. Maybe we need to change the CI workflow in a separate PR.

**Comment by mikemorency** [comment]
i tried adding the `extra-collections`. I can see the collection is installed, but it doesnt seem to have an affect for some reason.

Also theres an `ayncio` deprecation thats causing the python3.12 tests to fail. Im not sure what the best way to move forward with that is

**Comment by mariolenz** [comment]
This is weird, I don't see any docs checks ATM and I don't understand why...

**Comment by mariolenz** [comment]
@mikemorency How did you generate `config/api_specifications/8.0.2`? I think you should document this in this PR.

Not to put too fine a point on this, but I think there's a general problem here. You people are doing a lot of PRs without really explaining. For example #494 which really sucks.

I always try to add as much information as possible in order to help people understand what I'm trying to do. See #449 for an example.

It would be really great if you would add a little bit more information to your PRs.

cc @alinabuzachis @bardielle @jillr

**Comment by mikemorency** [comment]
I was using this draft PR to test CI results, so I didnt fill out the PR description. Ill do that now!

The steps I took to generate this release have been documented in `development.md` included in the PR. For the api spec:
```
1. Get the OpenAPI spec from VCenter
    a. Clone https://github.com/vmware/vmware-openapi-generator
    b. Run `python vmsgen.py -vc 'your_vcenter_ip_or_hostname' -o gen_out --insecure --oas 2` from within the checked out repo
    c. validate the directory `gen_out` has the spec files. Note `--oas 2` generates version 2 spec files. Version 3 seems bugged https://github.com/vmware/vmware-openapi-generator/issues/73
```

This experience has certainly made me feel that a general lack of documentation and "magic behind the scenes" makes contribution to this repo challenging.

**Comment by mariolenz** [comment]
> This experience has certainly made me feel that a general lack of documentation and "magic behind the scenes" makes contribution to this repo challenging.

Yes, that's exactly why I think it helps to be more explicit when creating a PR and tell people what has been done. This helps people to understand what has been done, even if the documentation is outdated, missing important steps  or even plain wrong.

Additionally, I like to document exactly what I did in a PR even for myself. It helps me if I want to do something similar again. I don't have to remember, I can look at my PRs and start from there ;-)

**Comment by mariolenz** [comment]
@mikemorency Cool! It looks like you were able to fix the docs CI job. Thanks, really great!

I'm not sure why the ansible-test-cloud-integration-vmware-rest Zuul job fails, though. I'll try to investigate.

**Review by mariolenz — APPROVED** [review]
LGTM

**Review by machacekondra — APPROVED** [review]

File: plugins/modules/appliance_access_dcui.py [hunk 1]
```diff
@@ -70,11 +70,13 @@
         type: bool
 author:
 - Ansible Cloud Team (@ansible-collections)
-version_added: 2.0.0
+version_added: 1.0.0
```
Comment (machacekondra) [inline plugins/modules/appliance_access_dcui.py:73]: This actually must be changed in all modules back to `version_added: 1.0.0`

**Review by bardielle — APPROVED** [review]

File: plugins/modules/appliance_access_dcui.py (hunk 1 above)
Comment (mikemorency) [reply plugins/modules/appliance_access_dcui.py:73]: all of the modules are set to `1.0.0` in the PR, did you mean change it back to `2.0.0`?

Comment (mikemorency) [reply plugins/modules/appliance_access_dcui.py]: ive restored the original version values, and added a playbook task to do that automatically next time

**Comment by mariolenz** [comment]
Looks like we have three approvals and no open change requests. I'll add the `mergeit` label.

@bardielle @machacekondra Please feel free to open an undo commit if you think I've been wrong merging this PR.

@mikemorency Maybe you should wait one or two days before you add the 4.0.0 tag that triggers the release. Just in case someone changes their mind.

**Comment by mariolenz** [comment]
I've merged this manually, but while the CI looked OK the `Publish Collection Docs / Build Ansible Docs (push)` failed. Any ideas why?

Any ideas why?

**Comment by mikemorency** [comment]
the push workflow [here](https://github.com/ansible-collections/vmware.vmware_rest/blob/main/.github/workflows/docs-push.yml) is missing this [attribute](https://github.com/ansible-collections/vmware.vmware_rest/blob/main/.github/workflows/docs-pr.yml#L26-L27)

**Pull request merged**